projects/angular/src/forms/datepicker/datepicker-view-manager.ts

```diff
@@ -52,7 +55,22 @@ export class ClrDatepickerViewManager {
     return this.viewManagerService.isDayView;
   }
 
+  get hasRangeOptions() {
+    return this.dateNavigationService && this.dateNavigationService.isRangePicker && this.dateRangeOptions?.length;
```

Comment:
```suggestion
    return this.dateNavigationService?.isRangePicker && this.dateRangeOptions?.length;
```

This line can be simplified.